source/code/plugin/health/health_monitor_optimizer.rb

```diff
@@ -0,0 +1,52 @@
+module HealthModel
+    class HealthMonitorOptimizer
+        #ctor
+        def initialize
+            @@health_signal_timeout = 240
+            @@first_record_sent = {}
+        end
+
+        def should_send(monitor_instance_id, health_monitor_state, health_monitor_config)
+
+            health_monitor_instance_state = health_monitor_state.get_state(monitor_instance_id)
+            health_monitor_records = health_monitor_instance_state.prev_records
+            health_monitor_config['ConsecutiveSamplesForStateTransition'].nil? ? samples_to_check = 1 : samples_to_check = health_monitor_config['ConsecutiveSamplesForStateTransition'].to_i
+
+            latest_record = health_monitor_records[health_monitor_records.size-1] #since we push new records to the end, and remove oldest records from the beginning
+            latest_record_state = latest_record["state"]
+            latest_record_time = latest_record["timestamp"] #string representation of time
+
+            new_state = health_monitor_instance_state.new_state
+            prev_sent_time = health_monitor_instance_state.prev_sent_record_time
+            time_first_observed = health_monitor_instance_state.state_change_time
+
+            if latest_record_state.downcase == new_state.downcase
+                time_elapsed = (Time.parse(latest_record_time) - Time.parse(prev_sent_time)) / 60
+                if time_elapsed > @@health_signal_timeout # minutes
```

Comment:
240 minutes ? what timeout is this used for ? can u pls explain ?

Reply:
this is the timeout when the state doesnt change. i.e. we send a signal every 4 hours if the state hasnt changed for a monitor